src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateArr = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+    const st = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties':
+        for (const key in action.extraData) {
+          st[key] = action.extraData[key];
+        }
+        break;
+      case 'removeProperties':
+        for (let j = 0; j < action.keysToRemove.length; j++) {
+          const key = action.keysToRemove[j];
+
+          delete st[key];
+        }
+        break;
+      case 'clear':
+        for (const key in st) {
+          delete st[key];
+        }
+        break;
+    }
```

Comment:
The switch statement lacks a default case. Adding a default case is crucial for handling unexpected action types, which will improve error handling and make the function more robust.